app.js

```diff
@@ -1,246 +1,97 @@
+function CookieStore (name, minCustomers, maxCustomers, avgCookies, hourlyCookieCount) { // this is a constructor function taking in 4 parameters.
   this.name = name || 'Unknown'; // adding || (or) inputs a value if it doesn't meet true.
   this.minCustomers = minCustomers;  // the context of this is going to change. The keyword 'this' knows what it represents by the surrounding context
   this.maxCustomers = maxCustomers; // argument on the right is going into the parameter in the function.
   this.avgCookies = avgCookies;
-  this.hourlyCount = hourlyCount || []; // assigning an empty array []
+  this.hourlyCookieCount = hourlyCookieCount || [];// assigning an empty array []
+  this.totalCountForDay = 0;
+  this.businessHours = ['07am ', '08am ', '09am ', '010am ', '011am ', '12pm ', '01pm ', '02pm ', '03pm ', '04pm ', '05pm ','06pm ','07pm ', '08pm '];
 }
 
 CookieStore.prototype.getAvgCookieCount = function () { // Add method with a prototype
-  var avgCookiesPerhour = Math.floor(Math.random() * ((this.maxCustomers - this.minCustomers + 1) + this.minCustomers) * this.avgCookies);
-  this.hourlyCount.push(avgCookiesPerhour);
-  console.log(this.name + 'Got avg cookie count.');
+  for (var hours = 0; hours < this.businessHours.length; hours++) {
+    var avgCookiesPerhour = Math.floor(Math.random() * ((this.maxCustomers - this.minCustomers + 1) + this.minCustomers) * this.avgCookies);
+    this.hourlyCookieCount.push(avgCookiesPerhour);
+    this.totalCountForDay += avgCookiesPerhour;
+  }
 };
 
-// var hourlyCookie = [50, 100, 23, 15, 66]; //Add array to your constructor as a parameter
+var firstAndPike = new CookieStore('First and Pike Store', 23, 65, 6.3); // CookieStore is a new type of instance.
+var seaTacAirport = new CookieStore('SeaTac Airport Store', 3, 24, 1.2); // Now, two new instances of CookieStore.
+var seattleCenter = new CookieStore('Seattle Center Store', 11, 38, 3.7);
+var capitolHill = new CookieStore('Capitol Hill Store', 20, 38, 2.3);
+var alki = new CookieStore('Alki Store', 2, 16, 4.6);
 
-var pikePlace = new CookieStore('Pike Place', 23, 65, 6.3); // CookieStore is a new type of instance.
-var otherStore = new CookieStore('Other Store', 100, 200, 10); // Now, two new instances of CookieStore.
-var anotherStore = new CookieStore('Downtown', 50, 500, 1600);
+firstAndPike.getAvgCookieCount();
+seaTacAirport.getAvgCookieCount();
+seattleCenter.getAvgCookieCount();
+capitolHill.getAvgCookieCount();
+alki.getAvgCookieCount();
 
-var stores = [pikePlace, otherStore, anotherStore];
+var stores = [firstAndPike, seaTacAirport, seattleCenter, capitolHill, alki];
 
-var tableEl = document.createElement('table');  //Step 1 - Create Element
+var storeSalesDataEl = document.createElement('table');
 
-for (var i = 0; i < stores.length; i++){
-  var currentStore = store[i];
+var timeRowEl = document.createElement('tr');
```

Comment:
from the lab instructions: the header row and footer row are each created in their own stand-alone function